src/main/java/com/netflix/control/clutch/ExperimentalControlLoop.java

```diff
@@ -34,35 +34,33 @@
 public class ExperimentalControlLoop implements Observable.Transformer<Event, Double> {
     private final ClutchConfiguration config;
     private final IActuator actuator;
-    private final Double initialSize;
     private final AtomicDouble dampener;
 
     private final AtomicLong cooldownTimestamp;
-    private final AtomicLong currentScale;
+    private final AtomicLong currentSize;
     private final AtomicDouble lastLag;
     private final Observable<Integer> size;
     private final IRpsMetricComputer rpsMetricComputer;
     private final IScaleComputer scaleComputer;
     private long cooldownMillis;
 
-    public ExperimentalControlLoop(ClutchConfiguration config, IActuator actuator, Double initialSize,
+    public ExperimentalControlLoop(ClutchConfiguration config, IActuator actuator, AtomicLong currentSize,
                                    Observable<Long> timer, Observable<Integer> size) {
-        this(config, actuator, initialSize, new AtomicDouble(1.0), timer, size,
+        this(config, actuator, currentSize, new AtomicDouble(1.0), timer, size,
                 new DefaultRpsMetricComputer(), new DefaultScaleComputer());
     }
 
-    public ExperimentalControlLoop(ClutchConfiguration config, IActuator actuator, Double initialSize,
+    public ExperimentalControlLoop(ClutchConfiguration config, IActuator actuator, AtomicLong currentSize,
                                    AtomicDouble dampener, Observable<Long> timer, Observable<Integer> size,
                                    IRpsMetricComputer rpsMetricComputer,
                                    IScaleComputer scaleComputer) {
         this.config = config;
         this.actuator = actuator;
-        this.initialSize = initialSize;
         this.dampener = dampener;
         this.cooldownMillis = config.getCooldownUnits().toMillis(config.cooldownInterval);
 
-        this.cooldownTimestamp = new AtomicLong(System.currentTimeMillis() + this.cooldownMillis);
-        this.currentScale = new AtomicLong(Math.round(initialSize));
+        this.cooldownTimestamp = new AtomicLong(System.currentTimeMillis());
```

Comment:
what was the cooldown doing before? waiting that many millis before making decisions?

Reply:
That's right. However, cooldown is already handled on line 112. So adding cooldown here was actually doubling cooldown initially.